rpcs/blockService.go

```diff
@@ -285,6 +291,27 @@ func (bs *BlockService) handleCatchupReq(ctx context.Context, reqMsg network.Inc
 	return
 }
 
+func (bs *BlockService) redirectRequest(round uint64, response http.ResponseWriter, request *http.Request) (err error) {
+
+	peers := bs.net.GetPeers(network.PeersPhonebookArchivers)
+	if len(peers) == 0 {
+		return fmt.Errorf("redirecRequest: no arcihver peers found")
+	}
+	peer := peers[0]
+	httpPeer, validHTTPPeer := peer.(network.HTTPPeer)
+	if !validHTTPPeer {
+		return fmt.Errorf("redirecRequest: error getting an http peer")
+	}
```

Comment:
use for-loop to find the first `network.HTTPPeer` within the peers array. Otherwise, you can return an error.